scripts/20170117-col629/rename_and_hide_suitec_folders.js

```diff
@@ -0,0 +1,164 @@
+          {'updated': updated, 'notFound': notFound, 'errored': errored},
+          'Finished. Updated folder in ' + updated.length + ' courses; folder not found in ' + notFound.length + ' courses; ' + errored.length + ' errors.'
+        );
+      }
+    });
+  });
+};
+
+// This function is our clunky means of leaving space between requests so as not to trigger Canvas throttling.
+var lastUpdate = Date.now();
+var delayedUpdateFolderForCourse = function(course, callback) {
+  var timeout = 5000 - (Date.now() - lastUpdate);
+  if (timeout < 0) {
+    timeout = 0;
+  }
+  lastUpdate = Date.now();
+  setTimeout(updateFolderForCourse, timeout, course, callback);
+};
+
+// Look for a _collabosphere folder. If one exists, rename it to _suitec and mark it as hidden. 
+var updateFolderForCourse = function(course, callback) {
+  log.debug({
+    'course': course.id,
+    'canvas_course_id': course.canvas_course_id,
+    'canvas_api_domain': course.canvas_api_domain
+  }, 'Updating folder for a Canvas course');
+
+  var getFolderUrl = util.format('%s/api/v1/courses/%d/folders/by_path/_collabosphere', getCanvasBaseURI(course.canvas), course.canvas_course_id);
+  var getFolderOpts = {
+    'url': getFolderUrl,
+    'method': 'GET',
+    'headers': {
+      'Authorization': util.format('Bearer %s', course.canvas.api_key)
+    }
+  };
+
+  request(getFolderOpts, function(err, response, body) {
+    if (err) {
+      log.error({'err': err, 'course': course.id}, 'Failed to check existence of Canvas upload folder');
+      errored.push(course.id);
+      return callback();
+    } else if (response.statusCode === 404) {
+      log.info({'course': course.id}, 'No _collabosphere folder found in course');
+      notFound.push(course.id);
+      return callback();
+    } else if (response.statusCode !== 200) {
+      log.error({'response': response, 'course': course.id}, 'Failed to check existence of Canvas upload folder; unexpected status code');
+      errored.push(course.id);
+      return callback();
+    }
+
+    var folderInfo = null;
+    try {
+      folderInfo = JSON.parse(body);
+    } catch (parseErr) {
+      log.error({'err': parseErr, 'course': course.id}, 'Failed to parse Canvas response');
+      errored.push(course.id);
+      return callback();
+    }
+
+    var folderId = _.last(folderInfo).id;
+    var updateFolderUrl = util.format('%s/api/v1/folders/%d', getCanvasBaseURI(course.canvas), folderId);
+    var updateFolderOpts = {
+      'url': updateFolderUrl,
+      'method': 'PUT',
+      'headers': {
+        'Authorization': util.format('Bearer %s', course.canvas.api_key)
+      },
+      'json': {
+      	'name': '_suitec',
+      	'hidden': true
+      }
+    };
+
+  	request(updateFolderOpts, function(err, response, body) {
+      if (err || (response.statusCode !== 200)) {
+        log.error({'err': err, 'response': response, 'course': course.id}, 'Failed to update folder in Canvas');
+      	errored++;
+      	return callback();
+      }
```

Comment:
@paulkerschen Sorry for not being more explicit... I believe an indentation fix is needed on all six lines above.